### PR TITLE
Updated checkpoint handling to enable zWGRv1 spending

### PR DESCRIPTION
**Pull request by Kokary**

### CODE REVIEW

**Review by ckti — COMMENTED** [review]
Looks like the code is a bit less cluttered as well

**Comment by Cryptarchist** [comment]
Code looks good, I will build and test the zWGRv1 spending stuff.

Nit outside the scope of this PR is whether the command should be named `calculateaccumulatorvalues` (<-- command name) or `generateaccumulatorvalues` (<-- help text name).

**Comment by Kokary** [comment]
No you're right, I updated the help wording.

(FYI, it's a helper function that I added to generate the json.h file, and introduces an alternative to the existing getaccumulatorvalues function: one that doesn't use existing checkpoints or the local database. Wasn't sure if we should keep it, but I figured: why not?)

**Comment by Cryptarchist** [comment]
@Kokary I think keep it. I played around with it earlier when I was testing this PR, took a few CPU cycles for it to calculate the values (thought it was broken), but it eventually spat out the matching results to match those in `accumulatorcheckpoints.json.h`.

File: src/accumulatorcheckpoints.json.h [hunk 1]
```diff
@@ -10,26 +10,345 @@
...
-            "    \"height\": 60,\n"
-            "    \"1\":    \"3c1\",\n"
-            "    \"5\":    \"3c1\",\n"
-            "    \"10\":   \"3c1\",\n"
-            "    \"50\":   \"3c1\",\n"
-            "    \"100\":  \"3c1\",\n"
-            "    \"500\":  \"3c1\",\n"
-            "    \"1000\": \"3c1\",\n"
-            "    \"5000\": \"3c1\"\n"
+            "    \"height\": 10020,\n"
```
Comment (Cryptarchist) [inline src/accumulatorcheckpoints.json.h:13]: The values for height `10020` seem to be duplicated on line 35.

File: src/accumulatorcheckpoints.json.h [hunk 2]
```diff
@@ -10,26 +10,345 @@
...
+            "    \"1000\": \"1ae6042c01118c9a1b75bd883001bb60dd53fe14c01804f8a0255edadb6618edd423e7847ef9ce144c2c4f4313f587d02cb546954c6503df202e96e05567e24d059ec4cb2f848cf31ecf8ddea7e88fb83e7ab845f1ed1703c927db917acc27c3ea72c66a84bcaf87feb6b946e1c42ca11008d21bfce1b202bb42de55d8ff42872e041065c616b34d42926820d3fadf41c5675ec6406506370424fb6dbf3b3be26402f37aea542328001c55945a1857a3e8a41406c6326b381c8633e97dab4a3ba1366a9c43c038c5f0ad2b1fcd30a9c8cfc0a6628b9af5941da503ba5f84d7df127d1d934ea0ac74692ee8c18133a920aef6e44892d70757f02fffdbf80eee8a\",\n"
+            "    \"5000\": \"154170d673be7e543e84107cf23d6a7d114a6f2b47701c414acb2a9c0c42e138a1e2d74cb2fe80b6c6a760a4025f2e024e3785cad1b0bc3346967d59c340fabe99406bb4adfc1b8d4423511641fe532ac685697e19ab9a96919f92470ba956a9f8d285185ec0f198f6700374d5e8ba69438b098849e85c374f744221dbe3a5360e6fef7bc2c737a1aacf65afb49e078288079d334dc71aae73e18059124587344dfeecde259d47aa0d075eb95ac3688bed1e6eb445bf8a904224d73407d9676033ba039d87dfdc8c2842eb8277fea96e7a9d744825b292c74e067352123ed173ee3e2e1b4919427b6cbcdb806224d806361e304069a8a7e7c78cbe1c47615f3c\"\n"
+            "  },\n"
+            "  {\n"
+            "    \"height\": 20020,\n"
```
Comment (Cryptarchist) [inline src/accumulatorcheckpoints.json.h:24]: The values for height `20020` seem to be duplicated on line 46.

**Comment by Kokary** [comment]
Thanks, removed duplicate values.

**Comment by Cryptarchist** [comment]
zWGRv1 spending confirmed working. ACK 58be8893fafa9005d9bf151df792af111368e2e0

**Pull request merged**